src/ImageProcessor.php

```diff
@@ -229,6 +234,16 @@ public function process(FileInterface $file): FileInterface
             throw TempFileCreationFailedException::withFilename($tempFile);
         }
 
+        $image = $this->imageManager->make($tempFile);
+        /*
+        $file = $file->withMetadata([
+            'width' => $this->image->getWidth(),
+            'height' => $this->image->getHeight(),
+        ]);
+        */
+        $file = $file->withMetadataKey('width', $image->getWidth())
+            ->withMetadataKey('height', $image->getHeight());
```

Comment:
Honestly this should not go into the processor. For example in our company we don't have that use case and it can be implemented in other places as well. If at all something is changed to archive this, then it should be a generic callback, similar to what is done in the storage service that features callbacks.

Reply:
Do you have an idea how to write this more generically?
Could the library provide a processor for this?

Reply:
As I just wrote in the other PR the main issue around operations like this one is the chicken-egg problem.

To get information from a file through php that was stored on a FTP server you'll have to download it first so you can operate on it, read, write, manipulate it and then re-upload it through the adapter.

So we need to figure out a way to determine if the file is available in a local environment for the processor or if you need to download it first. Downloading it first causes more issues: If more than one processor does this, you'll end up with a lot unnecessary down and uploads degrating performance and wasting bandwith.

I think these kind of operations should go into the application, because the applications knows when the file was uploaded successfully and when it is still available on the disk of the system it runs on. If we could solve the chicken-egg problem from the lib perspective I'm open to other solutions but overall it looks like this would be always application logic because the lib itself doesn't care about if the file is already existing somewhere or if it has not yet been store. The file object doesn't know about this state and I have right now no good idea how we could make it aware of that state.